lib/internal/bootstrap/node.js

```diff
@@ -319,6 +319,27 @@ ObjectDefineProperty(process, 'features', {
 }
 
 const { emitWarning, emitWarningSync } = require('internal/process/warning');
+const { getOptionValue } = require('internal/options');
+
+let kTypeStrippingMode = null;
+ObjectDefineProperty(process.features, 'typescript', {
+  __proto__: null,
+  get() {
+    if (kTypeStrippingMode === null) {
+      if (getOptionValue('--experimental-transform-types')) {
+        kTypeStrippingMode = 'transform';
+      } else if (getOptionValue('--experimental-strip-types')) {
+        kTypeStrippingMode = 'strip';
+      } else {
+        kTypeStrippingMode = undefined;
```

Comment:
I think it is good to just return `false` here like other features property hinting support disabled, instead of presese of the property with `undefined` value. 

```suggestion
        kTypeStrippingMode = false;
```